### PR TITLE
PEAR-1658: Fix for trapping return focus on modal close using ESC

### PR DESCRIPTION
**Pull request by paribartandhakal**
## Description
**Note**: The implementation for when the summary modal is closed is not in this PR. I have created a separate ticket for it: https://gdc-ctds.atlassian.net/browse/PEAR-1726 which will be worked on after the soft launch to minimize any side effects.

_**Please verify if all the modals open as before. I have checked as much as I remembered.**_
cc. @craigrbarnes @wteouchicago @cemile-ctds 
## Checklist
- [X] Added proper unit tests
- [ ] Left proper TODO messages for any remaining tasks
- [ ] Scanned for web accessibility with **aXe**, and mitigated or documented
      flagged issues

### CODE REVIEW

**Comment by wteouchicago** [comment]
These are the modals I looked at to make sure they still load/open:
- Save / Save-As
- Replace
- Send Feedback
- Delete cohort
- Import cohort
- Repository and Cohort Builder > Add a custom filter
- cDAVE > Customize bins
- create cohort buttons (the ones with the orange + icon)
- Manage Sets > Create Set 
- Save/Edit Gene/Mutation Set above tables
- various links to summary pages modals

Note:
I couldn't test the User Profile modal cos I have no permissions right now. After I logged in, it gave me a runtime error (prob. not related to this PR): 
Screenshot 2024-01-09 at 6 11 45 PM


Issues:
1. **Should fix**. When I use the keyboard to navigate to Save/Save-As and then select it, I cannot use tab to change my focus from the name field to the Cancel/Save buttons in the Save Cohort modal. Possibly I'm doing something wrong. This is happening also for:
- Cohort Bar > Custom Filters > Cases/Genes/Mutations 
- cDAVE > Customize Bins modal for changing the focus from the X
- Manage Sets > Create Set > Genes/Mutations.  My guess is it'll happening whenever the trigger element is a menu item.
- Save New Cohort button above the cases table

If we have to choose, I'd prefer that we skip making focus return to these menu items for now rather than have these modals not-kb-navigable. 

https://github.com/NCI-GDC/gdc-frontend-framework/assets/73256434/b098560b-538c-4fbe-b0c2-2daa81d6a1ea

2. the Save New Cohort button in cDAVE, when disabled, now looks...weird. Doesn't seem to be happening in develop.
Screenshot 2024-01-09 at 6 35 41 PM

3. Not sure if it's included in this branch, but the create cohort + buttons in tables don't get the focus returned to them once I close the modal.

4.  **Blocker**. I'm getting some infinite loop trying to save a gene set in Mutation Frequency. This isn't happening in develop.
- use an unfiltered cohort
- go to MF. wait for table to finish loading
- select Save/Edit Gene Set > Save as new gene set. note that the "Save top" value is 0, which is odd. I expect 716.
- save with some name
- note the infinite "Set has been saved" notification

Note it's similar for trying to save a mutation set.

**Comment by paribartandhakal** [comment]
> 4\. infinite loop

Following issues have been fixed:
1. Runtime error in User profile modal has been fixed. Needed a check for null data.
2. KB navigation issue for the modals have been fixed. I have employed different technique to return focus back to the dropdown button element (consulted with @cemile-ctds and @amlehman). This is not interfering with mouse click.
3. the Save New Cohort button in cDAVE while disabled looks like before
4. **The create cohort + buttons in tables focus issue has NOT been solved as this is related to the table component and the ticket has been created and put in the description above**
5. Infinite loop trying to save a gene/mutation set in Mutation Frequency has been fixed. 


Please verify and let me know 😃

File: packages/portal-proto/src/features/proteinpaint/GeneExpressionWrapper.tsx [hunk 1]
```diff
@@ -145,12 +145,14 @@ export const GeneExpressionWrapper: FC<PpProps> = (props: PpProps) => {
...
-        <SaveCohortModal // Show the modal, create a saved cohort when save button is clicked
-          onClose={() => setShowSaveCohort(false)}
-          filters={newCohortFilters}
-        />
-      )}
+
+      <SaveCohortModal // Show the modal, create a saved cohort when save button is clicked
+        onClose={() => setShowSaveCohort(false)}
+        opened={showSaveCohort}
+        // opened={showSaveCohort && newCohortFilters} check if newCohortFilters is needed
```
Comment (paribartandhakal) [inline packages/portal-proto/src/features/proteinpaint/GeneExpressionWrapper.tsx]: have to come back to this as it didn't feel right.
cc. @craigrbarnes

**Comment by wteouchicago** [comment]
All modals open as normal for me now. 

Verified the issues reported are fixed (except 4, which is for a separate ticket). 
- caveat: I now have permissions and I am not sure if I did before, so it's possible the test conditions are a bit different for the user profile modal.

BTW, kb focus after closing summary modals seems to work now for some of the modal buttons, e.g. in MF tables? yay? 🤷‍♀️ 

Issue
6. Critical. Something odd is happening in Mutation Freq. Use demo for ease. Maybe related to previous issue about Save/Edit gene/mutation sets.

Scenario A:
- save a gene set. but in the name, type in "1" and then save
- start saving another gene set. note that it prefills the "Name" field "1". It should have provided you a default name based on your filters (which should be "TCGA-LGG, true" in this case if you haven't changed any filtering)
- now in the name field, type in another char, e.g. "2" (so the name is "12"). note that when you type that char, it immediately saves. It should wait for you to click Save before trying to save.
- this happens to mutation sets as well.

Scenario B:
- go to Mutations tab and try to save a mutation set
- note that the "Save top" field is blank. It should be prefilled (prob. with 2556 if on qa-orange)
- if I fill it with something, the next time I use the save mutation set option, it prefills with the value I typed in previously. 

Scenario C.
- Seems similar to A.
- When I use Add to or remove from existing gene set and click on a radio button in the table in the modal, it immediately saves. This doesn't seem to happen all the time -- it only seems to happen after I get scenario A to trigger. 😕 
- i was only able to get this to happen for gene sets.

File: packages/portal-proto/src/features/cDave/CDaveCard/CDaveCard.unit.test.tsx [hunk 2]
```diff
@@ -121,10 +133,17 @@ describe("CDaveCard", () => {
   });
 
   it("continuous result with data", () => {
-    jest.spyOn(core, "useCoreSelector").mockReturnValue({
-      field: "exposures.cigarettes_per_day",
-      type: "long",
-    });
+    // jest.spyOn(core, "useCoreSelector").mockReturnValue({
```
Comment (amlehman) [inline packages/portal-proto/src/features/cDave/CDaveCard/CDaveCard.unit.test.tsx]: Commented code

File: packages/portal-proto/src/features/proteinpaint/GeneExpressionWrapper.tsx (hunk 1 above)
Comment (amlehman) [reply packages/portal-proto/src/features/proteinpaint/GeneExpressionWrapper.tsx]: can you add a "TODO" to this comment?

File: packages/portal-proto/src/features/cDave/CDaveCard/CDaveCard.unit.test.tsx (hunk 2 above)
Comment (paribartandhakal) [reply packages/portal-proto/src/features/cDave/CDaveCard/CDaveCard.unit.test.tsx]: removed

**Comment by wteouchicago** [comment]
Using prod instead of qa-int due to some cert date error I'm having with qa-int when saving cohorts.

Checked through all the modals I previously listed. Also verified all previously listed issues are fixed.

__
## Newly found issues
7. Getting endless cohort saved confirmations. This isn't happening on develop for me. Steps:
- Liver cohort from home page
- Save ("Liver")
- Save-As, name it "Liver", Replace
- notice several "Liver has been saved..." notifications keeps showing up
- seems to happen whenever I use Replace

**Comment by paribartandhakal** [comment]
> Using prod instead of qa-int due to some cert date error I'm having with qa-int when saving cohorts.
> 
> Checked through all the modals I previously listed. Also verified all previously listed issues are fixed.
> 
> __
> 
> ## Newly found issues
> 7. Getting endless cohort saved confirmations. This isn't happening on develop for me. Steps:
> 
> * Liver cohort from home page
> * Save ("Liver")
> * Save-As, name it "Liver", Replace
> * notice several "Liver has been saved..." notifications keeps showing up
> * seems to happen whenever I use Replace

This issue has been fixed. 😃

File: packages/portal-proto/src/components/Modals/SaveCohortModal.tsx [hunk 3]
```diff
@@ -198,6 +201,7 @@ const SaveCohortModal = ({
 
         // Need to wait for request removing outdated cohorts to finish when replacing cohort
         if (replace) {
+          console.log("reached hereeeeee!");
```
Comment (wteouchicago) [inline packages/portal-proto/src/components/Modals/SaveCohortModal.tsx]: log :P

**Comment by wteouchicago** [comment]
looks good. no longer getting endless confirmations. 👍 🔁 👍

File: packages/portal-proto/src/components/Modals/SetModals/SaveSelectionModal.tsx [hunk 4]
```diff
@@ -62,43 +64,31 @@ const SaveSelectionAsSetModal: React.FC<SaveSelectionAsSetModalProps> = ({
...
-    dispatch,
-    setType,
-    closeModal,
-    form.values.name,
-  ]);
+    // form keeps on changing
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [opened, initialSetName, max]);
+
+  console.log({
```
Comment (amlehman) [inline packages/portal-proto/src/components/Modals/SetModals/SaveSelectionModal.tsx]: Log statement

File: packages/portal-proto/src/components/Modals/SetModals/SaveSelectionModal.tsx [hunk 5]
```diff
@@ -62,43 +64,31 @@ const SaveSelectionAsSetModal: React.FC<SaveSelectionAsSetModalProps> = ({
...
-    response.isError,
-    response.data,
-    dispatch,
-    setType,
-    closeModal,
-    form.values.name,
-  ]);
+    // form keeps on changing
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [opened, initialSetName, max]);
```
Comment (amlehman) [inline packages/portal-proto/src/components/Modals/SetModals/SaveSelectionModal.tsx]: Could you try wrapping this: `form.setValues((prev) => ({ ...prev, name: initialSetName, top: max }));` in a memoized callback and see if that solves the dependency issue here?

Comment (paribartandhakal) [reply packages/portal-proto/src/components/Modals/SetModals/SaveSelectionModal.tsx]: This is done. Thanks to you 😄

**Review by amlehman — APPROVED** [review]

**Review by craigrbarnes — APPROVED** [review]

**Pull request merged**